### DISCUSSION TITLE
Clipping a netcdf file with Shapefile in Anaconda produces NaN values

### DISCUSSION THREAD

**Discussion by MayMonThetKhine** [corteva/rioxarray#520] [Q&A]
I want to clip my netcdf file with the shapefile, the script I run is as below. But my clipped data when I check with Panoply gives the dtr values all NaN. I attached the screenshot here.

Please enlighten me on how I can solve this one and where I went wrong. Thanks much.

==================================================================================

import xarray
import rioxarray
import geopandas
from shapely.geometry import mapping

xds = xarray.open_dataset("dtr_MON_climpact.sample_historical_NA_1970-1989.nc", drop_variables='time_bnds')
xds.rio.write_crs("EPSG:4326", inplace=True)
geodf = geopandas.read_file("Chauk_BD.shx")

clipped = xds.rio.clip(geodf.geometry.apply(mapping), geodf.crs,drop=False)
clipped.to_netcdf('my clipped data 1.nc')
![NaN values](https://user-images.githubusercontent.com/104421747/166859190-0c7de3c0-99c9-4f55-ac3a-7c8cee629af0.png)

**Comment by JessicaS11** [comment]
> geodf = geopandas.read_file("Chauk_BD.shx")

Perhaps a typo that you're bringing in the .shx file instead of the .shp file?
